lib/pages/authen/register/widgets/input_form.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:structure_flutter/bloc/blocs/auth_provider.dart';
+import 'package:structure_flutter/core/common/constants/size_constant.dart';
+import 'package:structure_flutter/core/resource/app_colors.dart';
+import 'package:structure_flutter/core/resource/text_style.dart';
+
+class InputForm extends StatefulWidget {
+  @override
+  _InputFormState createState() => _InputFormState();
+}
+
+class _InputFormState extends State<InputForm> {
+  final TextEditingController _emailController = TextEditingController();
+  final TextEditingController _passwordController = TextEditingController();
+  final TextEditingController _confirmPasswordController =
+      TextEditingController();
+
+  AuthProvider _auth;
+
+  GlobalKey<FormState> _formKey;
+
+  @override
+  void initState() {
+    _formKey = GlobalKey<FormState>();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _emailController.dispose();
+    _passwordController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _auth = Provider.of<AuthProvider>(context);
+    SizeConstant().init(context);
+    return Container(
+      height: SizeConstant.screenHeight * 0.75,
+      child: Form(
+        key: _formKey,
+        onChanged: () {
+          _formKey.currentState.save();
+        },
+        child: Column(
+          children: [
+            Container(
+              padding: EdgeInsets.all(10.0),
+              alignment: Alignment.topLeft,
+              child: Text(
+                'Email',
+                style: black38_16,
+              ),
+            ),
+            TextFormField(
+              controller: _emailController,
+              validator: (_input) {
+                return _input.length != 0 && _input.contains("@")
```

Comment:
```suggestion
                return _input.isNotEmpty && _input.contains("@")
```
and what happen when i enter ```@@@@@@@@``` in email fields =))))

Reply:
💯  i just check basic case . Can i use "email_validator" dependencies to validate the mail in this case ?